cranelift/codegen/src/isa/x86/fde.rs

```diff
@@ -113,21 +122,34 @@ fn map_reg(isa: &dyn TargetIsa, reg: RegUnit) -> Register {
         X86_64::XMM14,
         X86_64::XMM15,
     ];
+
     let reg_info = isa.register_info();
-    let bank = reg_info.bank_containing_regunit(reg).unwrap();
+    let bank = reg_info
+        .bank_containing_regunit(reg)
+        .ok_or_else(|| RegisterMappingError::MissingBank)?;
     match bank.name {
         "IntRegs" => {
             // x86 GP registers have a weird mapping to DWARF registers, so we use a
             // lookup table.
-            X86_GP_REG_MAP[(reg - bank.first_unit) as usize]
-        }
-        "FloatRegs" => X86_XMM_REG_MAP[(reg - bank.first_unit) as usize],
-        _ => {
-            panic!("unsupported register bank: {}", bank.name);
+            Ok(X86_GP_REG_MAP[(reg - bank.first_unit) as usize])
         }
+        "FloatRegs" => Ok(X86_XMM_REG_MAP[(reg - bank.first_unit) as usize]),
+        _ => Err(RegisterMappingError::UnsupportedRegisterBank(
+            bank.name.into(),
+        )),
     }
 }
 
+#[derive(Error, Debug)]
+pub enum RegisterMappingError {
+    #[error("unable to find bank for register info")]
+    MissingBank,
+    #[error("register mapping is currently only implemented for x86_64")]
+    UnsupportedArchitecture,
+    #[error("unsupported register bank: {0}")]
+    UnsupportedRegisterBank(String),
```

Comment:
This could probably take a `&'static str`?

Reply:
Yes, let's make it `&'static str`.